### PR TITLE
Sha256 algo in checkjs

### PR DESCRIPTION
**Pull request by ArtemAnoshin**

### CODE REVIEW

File: cleantalk/cleantalk.php [hunk 1]
```diff
@@ -8,7 +8,8 @@
 		return;
 	}
 
-	$apbct_checkjs_val = md5($apikey);
+    global $salt;
```
Comment (Glomberg) [inline cleantalk/cleantalk.php]: Please add apbct_ prefix to the $salt variable.

**Review by Glomberg — CHANGES_REQUESTED** [review]

File: cleantalk/cleantalk.php (hunk 1 above)
Comment (ArtemAnoshin) [reply cleantalk/cleantalk.php]: Done

File: cleantalk/inc/functions.php [hunk 2]
```diff
@@ -446,9 +446,12 @@ function apbct_obfuscate_param($value = null)
 	 * return null|0|1;
 	 */
 	 function apbct_js_test(){
-		 global $apikey, $salt;
+		 global $apikey, $salt, $detected_cms;
 		 if(isset($_COOKIE['apbct_checkjs'])){
-			if($_COOKIE['apbct_checkjs'] == apbct_checkjs_hash($apikey, $salt))
+			if(
+                $_COOKIE['apbct_checkjs'] == apbct_checkjs_hash($apikey, $salt) ||
+                ($detected_cms === 'cscart' && $_COOKIE['apbct_checkjs'] == md5($apikey))
```
Comment (safronik) [inline cleantalk/inc/functions.php:453]: Why do we need this?

Comment (ArtemAnoshin) [reply cleantalk/inc/functions.php:453]: For compatibility

**Pull request merged**